### PR TITLE
Fix nonexistent secret update

### LINKED ISSUE #163: [BUG] Secret gets never populated if KeyVault entry was created after the secret template
[X] Controller, version: `1.2.0` (spvest/azure-keyvault-controller:1.2.0)
[ ] Env-Injector (webhook), version: `x.x.x` (docker image tag)
[ ] Other

**Describe the bug**
If secret template has been created before relevant Azure KeyVault entry it will not be populated even after KeyVault entry creation.

**To Reproduce**
1. Create secret template which references non-existent KeyVaul entry (see Manifest below). 
2. Create KeyVault entry.
3. Relevant k8s secret will not be created.

**Expected behavior**
Secret created once KeyVault controller receives update about the KeyVault entry referenced in the template

**Logs**
<details><summary>Logs</summary>
<p>

```
I0317 17:39:22.738719       1 azureKeyVaultSecret.go:229] "updating with recent changes from azure key vault" azurekeyvaultsecret="app-namespace/some-token" secret="app-namespace/some-token"
E0317 17:39:23.116356       1 worker.go:92] Failed to process key app-namespace/some-token. Reason: failed to get existing secret some-token, error: secrets "some-token" not found
I0317 17:43:56.989218       1 azureKeyVaultSecret.go:229] "updating with recent changes from azure key vault" azurekeyvaultsecret="app-namespace/some-token" secret="app-namespace/some-token"
E0317 17:43:57.316122       1 worker.go:92] Failed to process key app-namespace/some-token. Reason: failed to get existing secret some-token, error: secrets "some-token" not found
E0317 17:43:57.316168       1 worker.go:106] failed to get existing secret some-token, error: secrets "some-token" not found
I0317 17:48:21.989052       1 azureKeyVaultSecret.go:229] "updating with recent changes from azure key vault" azurekeyvaultsecret="app-namespace/some-token" secret="app-namespace/some-token"
E0317 17:48:22.316220       1 worker.go:92] Failed to process key app-namespace/some-token. Reason: failed to get existing secret some-token, error: secrets "some-token" not found

```
</p>
</details>

<details><summary>Manifest</summary>

```yaml
apiVersion: spv.no/v1
kind: AzureKeyVaultSecret
metadata:
  name: some-token
spec:
  output:
    secret:
      dataKey: some.token
      name: some-token
  vault:
    name: KeyVaultName
    object:
      name: app-namespace-some-token
      type: secret
```
</details>

**Additional context**
According to the logs controller assumes that secret has been already created and tries to find it to perform an update, but secret wasn't created. I believe controller should fall back to creating a secret if it was not found.

### PR DESCRIPTION
**Pull request by yix**
Fixes #163 (updated)

### CODE REVIEW

File: cmd/azure-keyvault-controller/controller/azureKeyVaultSecret.go [hunk 1]
```diff
@@ -229,21 +229,28 @@ func (c *Controller) syncAzureKeyVault(key string) error {
 			klog.InfoS("updating with recent changes from azure key vault", "azurekeyvaultsecret", klog.KObj(akvs), "secret", klog.KRef(akvs.Namespace, akvs.Spec.Output.Secret.Name))
 			existingSecret, err := c.kubeclientset.CoreV1().Secrets(akvs.Namespace).Get(context.TODO(), akvs.Spec.Output.Secret.Name, metav1.GetOptions{})
 			if err != nil {
-				return fmt.Errorf("failed to get existing secret %s, error: %+v", akvs.Spec.Output.Secret.Name, err)
-			}
+				klog.InfoS("failed to get existing secret %s, error: %+v", akvs.Spec.Output.Secret.Name, err)
```
Comment (181192) [inline cmd/azure-keyvault-controller/controller/azureKeyVaultSecret.go:232]: Use the `klog.ErrorS` if this is an error meant for stderr. `InfoS` doesn't support formatting.
```go
klog.ErrorS(err, "failed to get existing secret", "secret", akvs.Spec.Output.Secret.Name)
```

**Review by 181192 — COMMENTED** [review]
Looks good to me, just a minor comment. Will test after :smile:

File: cmd/azure-keyvault-controller/controller/azureKeyVaultSecret.go (hunk 1 above)
Comment (yix) [reply cmd/azure-keyvault-controller/controller/azureKeyVaultSecret.go]: Updated with `klog.Infof` to enable formatting. I believe informational message is more appropriate in this case.

**Comment by 181192** [comment]
Beautiful, working as expected, thanks! :tada:

**Comment by 181192** [comment]
@yix Released controller 1.2.2 included in Helm chart 2.0.8 😊

**Pull request merged**